infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -335,17 +338,146 @@ public void setSnapshotAppliedComplete(LogReplicationEntry entry) {
+    public void updateSnapshotSyncInfo(String clusterId) {
+        Instant time = Instant.now();
+        Timestamp timestamp = Timestamp.newBuilder().setSeconds(time.getEpochSecond())
+                .setNanos(time.getNano()).build();
+
+        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setClusterId(clusterId).build();
+        CorfuRecord<ReplicationStatusVal, Message> record =
+                corfuStore.query(NAMESPACE).getRecord(REPLICATION_STATUS_TABLE, key);
+        if (record != null) {
+            ReplicationStatusVal previous = record.getPayload();
+            LogReplicationMetadata.SnapshotSyncInfo previousSyncInfo = previous.getSnapshotSyncInfo();
+
+            LogReplicationMetadata.SnapshotSyncInfo currentSyncInfo = previousSyncInfo.toBuilder()
+                    .setStatus(LogReplicationMetadata.SyncStatus.COMPLETED)
+                    .setCompletedTime(timestamp)
+                    .build();
+
+            ReplicationStatusVal current = ReplicationStatusVal.newBuilder()
+                    .setRemainingEntriesToSend(previous.getRemainingEntriesToSend())
+                    .setSyncType(ReplicationStatusVal.SyncType.LOG_ENTRY)
+                    .setStatus(LogReplicationMetadata.SyncStatus.ONGOING)
+                    .setSnapshotSyncInfo(currentSyncInfo)
+                    .build();
+
+            corfuStore.tx(NAMESPACE)
+                    .update(REPLICATION_STATUS_TABLE, key, current, null)
+                    .commit();
+
+            log.debug("updateSnapshotSyncInfo as completed: clusterId: {}, syncInfo: {}",
+                    clusterId, currentSyncInfo);
+        }
+    }
+
+    /**
+     * Set replication status table.
+     * If the current sync type is log entry sync, keep Snapshot Sync Info.
+     *
+     * @param clusterId standby cluster id
+     * @param remainingEntries num of remaining entries to send
+     * @param type sync type
+     */
+    public void setReplicationStatusTable(String clusterId, long remainingEntries,
+                                          ReplicationStatusVal.SyncType type) {
+        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setClusterId(clusterId).build();
+        LogReplicationMetadata.SnapshotSyncInfo syncInfo = null;
+        ReplicationStatusVal current;
+
+        CorfuRecord<ReplicationStatusVal, Message> record =
+                corfuStore.query(NAMESPACE).getRecord(REPLICATION_STATUS_TABLE, key);
+        if (record != null) {
+            ReplicationStatusVal previous = record.getPayload();
+            syncInfo = previous.getSnapshotSyncInfo();
+        }
+
+        if (type == ReplicationStatusVal.SyncType.LOG_ENTRY) {
+            if (syncInfo == null){
+                log.warn("setReplicationStatusTable during LOG_ENTRY sync, " +
+                        "previous status is not present for cluster: {}", clusterId);
+                syncInfo = LogReplicationMetadata.SnapshotSyncInfo.newBuilder().build();
```

Comment:
In the case it was not set and we're coming from negotiation saying we can continue with LOGENTRY, on negotiation don't we get info of the last baseSnapshotTimestamp that at least we could populate? Maybe it's not worth the work but something to consider, what do you think? or we can leave a todo..

Reply:
I don't think syncInfo is null if we come from negotiation. It should have a previous round snapshot sync, and syncInfo should be kept.